### PR TITLE
WRP-565: Refine custom event dispatching

### PR DESCRIPTION
**Pull request by 0x64**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Some custom events are based on native user inputs like keyboard, mouse, or touch, and it would be helpful for apps to deal with scenarios that we provide `stopPropagation` and `preventDefault` in custom events' payloads.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Fixed `core/handle/forwardCustomWithPrevent` not to share its closure variable to handlers
- Added `stopPropagation` and `preventDefault` to some custom events
- Refined code forwarding events

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-565

### Comments
Enact-DCO-1.0-Signed-off-by: Seungcheon Baek (sc.baek@lge.com)

### CODE REVIEW

File: packages/core/handle/tests/handle-specs.js [hunk 1]
```diff
@@ -567,4 +568,106 @@ describe('handle', () => {
 			expect(actual).toEqual(expected);
 		});
 	});
+
+	describe('#forwardCustomWithPrevent', () => {
+		test('should pass an object with `type` when no adapter is provided', () => {
```
Comment (MikyungKim) [inline packages/core/handle/tests/handle-specs.js:573]: We are checking the `preventDefault` function too. Please add a description for this.

File: packages/core/handle/tests/handle-specs.js [hunk 2]
```diff
@@ -567,4 +568,106 @@ describe('handle', () => {
...
+			];
+			forwardCustomWithPrevent('onCustomEvent', adapter)(...args);
+
+			const expected = args;
+			const actual = adapter.mock.calls[0];
+
+			expect(actual).toEqual(expected);
+		});
+
+		test('should events to function specified in provided props when preventDefault() hasn\'t been called', () => {
```
Comment (MikyungKim) [inline packages/core/handle/tests/handle-specs.js:650]:
```suggestion
		test('should call the next handler when `preventDefault` from provided props hasn\'t been called', () => {
```

File: packages/core/handle/tests/handle-specs.js [hunk 3]
```diff
@@ -567,4 +568,106 @@ describe('handle', () => {
...
+			const event = 'onMyClick';
+			const handler = jest.fn();
+
+			const callback = handle(forwardCustomWithPrevent(event), handler);
+
+			callback();
+			expect(handler).toHaveBeenCalledTimes(1);
+		});
+
+		test('should not events to function specified in provided props when preventDefault() has been called', () => {
```
Comment (MikyungKim) [inline packages/core/handle/tests/handle-specs.js:660]:
```suggestion
		test('should not call the next handler when `preventDefault` from provided props has been called', () => {
```

File: packages/core/handle/tests/handle-specs.js (hunk 1 above)
Comment (0x64) [reply packages/core/handle/tests/handle-specs.js]: ✅ I added a description for the next test also. 😉

File: packages/core/handle/tests/handle-specs.js (hunk 2 above)
Comment (0x64) [reply packages/core/handle/tests/handle-specs.js]: 👍 ✅

File: packages/core/handle/tests/handle-specs.js (hunk 3 above)
Comment (0x64) [reply packages/core/handle/tests/handle-specs.js]: 👍 ✅

**Review by MikyungKim — APPROVED** [review]
LGTM

**Review by juwonjeong — APPROVED** [review]
LGTM

**Pull request merged**